registry/layer.go

```diff
@@ -59,7 +59,7 @@ func (lh *layerHandler) GetLayer(w http.ResponseWriter, r *http.Request) {
 	defer layer.Close()
 
 	if lh.layerHandler != nil {
-		handler, _ := lh.layerHandler.Resolve(layer)
+		handler, _ := lh.layerHandler.Resolve(layer, map[string]interface{}{"method": r.Method})
```

Comment:
Aren't these options also possible passed through configuration? I'm not sure that this is the right approach.